File: client/src/app/models/helpers/queryParameters.ts
import { HttpParams } from '@angular/common/http';

export class QueryParameters {
  pageNumber: number = 1;
  pageSize: number = 10;
  searchTerm: string = '';

  getQueryString() {
    let queryString = '';
    queryString += this.pageNumber + '-';

    queryString += this.pageSize;

    return queryString;
  }

  getHttpParamsObject() {
    let params = new HttpParams();
    params = params.append('pageNumber', this.pageNumber);

    params = params.append('pageSize', this.pageSize);

    if (this.searchTerm) {
      params = params.append('searchTerm', this.searchTerm);
    }

    return params;
  }
}

File: API/Helpers/QueryParameters.cs
namespace API.Helpers;

public class QueryParameters
{
    public string? SearchTerm { get; set; }
    
    
    private int MaxPageSize = 48;

    public int PageNumber { get; set; } = 1;
    
    private int _pageSize = 48;
    public int PageSize
    {
        get => _pageSize;
        set => _pageSize = (value > MaxPageSize) ? MaxPageSize : value;
    }
}